app/controllers/catalog_controller.rb

```diff
@@ -0,0 +1,40 @@
+require 'moab/stanford'
+
+##
+# CatalogController allows clients to manipulate the Object Inventory Catalog, e.g.
+# to add an existing moab object to the catalog, or to update an entry for a moab object
+# that's already in the catalog.
+class CatalogController < ApplicationController
+  before_action :set_id
+
+  def add_preserved_object
+    PreservedObject.create!(
+      druid: @id, size: moab_size(@id), current_version: Stanford::StorageServices.current_version(@id)
+    )
+  end
+
+  def update_preserved_object
+    preserved_obj = PreservedObject.find_by(druid: @id)
+    preserved_obj.update_attributes(
+      druid: @id, size: moab_size(@id), current_version: Stanford::StorageServices.current_version(@id)
+    )
+  end
+
+  private
+
+  def set_id
+    @id = catalog_params[:id]
+    head(:unprocessable_entity) if @id.blank?
+  end
+
+  def catalog_params
+    params.permit(:id)
```

Comment:
Is it worth using `require()` here?

Reply:
if you do a `require`, you're required to have the specific params you permit in a hash named by the `require`.  e.g., if you do `params.require(:catalog_params).permit(:id)`, the callers are required to pass in a hash of params called `catalog_params`, and that hash can contain a field called `id`.  this seems more useful when you have a form to submit, and where the rails form builder stuff is handling that construction for you (using, IIRC, named forms, which will build that hash into the form structure).  since (IME) all our usage of strong params is with form submission, and since this is what the majority of examples i've seen online are dealing with, we're used to always seeing the `require` part.

but that seems overly circuitous for a REST call that's just going to be providing some parameters without being in the context of a form.  we just want to make sure that we only allow specific param names, but we don't really need to bother having them in a hash.  the rails docs do actually have instructions for this, they're just not the first thing described:
http://edgeguides.rubyonrails.org/action_controller_overview.html#permitted-scalar-values (section 4.5.1, my browser landed just past the anchor for that section when i reloaded that link, so you may want to scroll up if it doesn't land properly on that section).

we did originally implement it using `require`, but then the request in the unit tests looked like:
```
post :add_preserved_object, params: { catalog_params: { id: id } }
```

instead of:
```ruby
post :add_preserved_object, params: { id: id }
```

and we thought the latter would be preferable for real callers.